### PR TITLE
BUG: catch decimal.InvalidOperation exception from dateutil

### LINKED ISSUE #51084: BUG: to_datetime with errors coerce throws InvalidOperation decimal.DivisionImpossible
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
x = pd.to_datetime('87156549591102612381000001219H5',infer_datetime_format=True, errors='coerce')
```

### Issue Description

Function should return NaT in case of problem, not exception decimal.DivisionImpossible:
InvalidOperation: [<class 'decimal.DivisionImpossible'>]

Problem present also in former versions of pandas.

### Expected Behavior

x = pd.NaT

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 87cfe4e38bafe7300a6003a1d18bd80f3f77c763
python           : 3.10.4.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 154 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : English_United States.1252

pandas           : 1.5.0
numpy            : 1.23.1
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 57.0.0
pip              : 22.2.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.31.1
pandas_datareader: None
bs4              : None
bottleneck       : 1.3.5
brotli           : 
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.3
numba            : None
numexpr          : 2.8.3
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by kcaashish**
- [X] closes #51084  
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [X] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/parsing.pyx [hunk 1]
```diff
@@ -646,7 +648,10 @@ cdef datetime dateutil_parse(
         str reso = None
         dict repl = {}
 
-    res, _ = DEFAULTPARSER._parse(timestr, dayfirst=dayfirst, yearfirst=yearfirst)
+    try:
+        res, _ = DEFAULTPARSER._parse(timestr, dayfirst=dayfirst, yearfirst=yearfirst)
+    except InvalidOperation:
+        res = None
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/parsing.pyx:654]: is there a test which hits this? if not, could you add one please, else someone might remove it thinking it's not needed

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @kcaashish ! This is off to a good start

File: pandas/_libs/tslibs/parsing.pyx (hunk 1 above)
Comment (kcaashish) [reply pandas/_libs/tslibs/parsing.pyx:654]: I will look into it. Thank you!

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/parsing.pyx:654]: I think something like `to_datetime(['87156549591102612381000001219H5'])` might be enough to hit it, could you put such an example in `test_to_datetime`?

Comment (kcaashish) [reply pandas/_libs/tslibs/parsing.pyx:654]: Updated as per the comments.

File: pandas/tests/tools/test_to_datetime.py [hunk 2]
```diff
@@ -2501,6 +2501,16 @@ def test_to_datetime_overflow(self):
         with pytest.raises(OutOfBoundsTimedelta, match=msg):
             date_range(start="1/1/1700", freq="B", periods=100000)
 
+    def test_string_invalid_operation(self, cache):
```
Comment (kcaashish) [inline pandas/tests/tools/test_to_datetime.py:2504]: I didn't know if I should have combined it with  `test_string_na_nat_conversion_malformed`, so I created a different function.

**Review by MarcoGorelli — APPROVED** [review]
Nice one, thanks @kcaashish ! lgtm pending green

**Comment by kcaashish** [comment]
Thank you!

**Review by jbrockmendel — COMMENTED** [review]
LGTM

**Pull request merged**